contracts/access/AccessControlDefaultAdminRules.sol

```diff
@@ -207,9 +267,30 @@ abstract contract AccessControlDefaultAdminRules is IAccessControlDefaultAdminRu
      * Internal function without access restriction.
      */
     function _beginDefaultAdminTransfer(address newAdmin) internal virtual {
-        _defaultAdminTransferDelayedUntil = SafeCast.toUint48(block.timestamp) + defaultAdminDelay();
+        uint48 delayChangeSchedule = defaultAdminDelayChangeSchedule();
+
+        if (delayChangeSchedule != 0) {
+            require(delayChangeSchedule < block.timestamp, "AccessControl: Delay change in progress");
+            // Must read the raw private value to avoid getting a 0 from `pendingDefaultAdminDelay()`
+            _currentDefaultAdminDelay = _pendingDefaultAdminDelay;
+            _resetDefaultAdminDelayChange();
+        }
```

Comment:
This is key logic to compute the correct admin delay and update the storage variables that we are placing in a function that deals with a different concern, which is transferring the admin. We should separate these concerns, and design internal abstractions so that this function can focus on the admin transfer.

With the changes I've suggested in other comments it should not be necessary to update the storage variables. We should be able to replace these lines with just `require(defaultAdminDelayChangeSchedule() == 0)`.

Reply:
I think this is now solved since the `_resetDefaultAdminDelayChange` is treated now as a noop, materializing the value new delay into storage whenever is needed.

Can you take a look, I think it may be cleaner?